File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/decorator/uielement/UIDecorator.java
// UIDecorator.java

package com.bigboxcode.designpattern.decorator.uielement;

public abstract class UIDecorator implements UIElement {
    protected final UIElement uiElement;

    protected UIDecorator(UIElement uiElement) {
        this.uiElement = uiElement;
    }

    @Override
    public void draw() {
        this.uiElement.draw();
    }
}

File: design-patterns/php/decorator/uielement/UiDecorator.php
<?php
// UiDecorator.php

namespace BigBoxCode\DesignPattern\Decorator\UiElement;

abstract class UIDecorator implements UIElement {
    protected function __construct(protected readonly UIElement $uiElement) {
    }

    public function draw(): void {
        $this->uiElement->draw();
    }
}